chroma-core/chroma
# fix: wrong cjs/esm builds

## Description of changes
*Summarize the changes made by this PR.*
 - Improvements & Bug fixes
	 - implement tsup to create cjs and esm builds and declare chromadb as module
	   -  resolves #1123

## Test plan
*How are these changes tested?*

- run `npx verdaccio`
- create the file `clients/js/.npmrc` with content `registry=http://localhost:4873`
- increase the package version
- run `npm publish`
- start a clean js project
   - place the same  .npmrc here
   - install chromadb `yarn add chromadb` or `npm install chromadb`
   - create a index.js with content `const { ChromaClient } = require('chromadb'); new ChromaClient().heartbeat();`
   - run `node index.js`
   - you should see an error that chroma wasn't able to connect - this means the chromadb client is working ✅ cjs is working
     - if your local chroma instance is running, you won't see any errors
   - add `"type":"module"` to the package.json
   - change the content of index.js to `import { ChromaClient } from 'chromadb'; new ChromaClient().heartbeat();`
   - you should see an error that chroma wasn't able to connect - this means the chromadb client is working ✅ esm is working
     - if your local chroma instance is running, you won't see any errors

## Documentation Changes
*Are all docstrings for user-facing APIs updated if required? Do we need to make documentation changes in the [docs repository](https://github.com/chroma-core/docs)?*
